### ISSUE TITLE
Actual State of development and Roadmap

### ISSUE THREAD

**Issue by art-cg** [mtarld/api-platform-ms-bundle#54]
Hi,
is there a Roadmap of planned features or any infos about actual state of development?
I like this bundle.
I just guess the community which uses API-Platform as Microservices is very small.
If someone wants to share their experiences, i would be glad to hear them.

Greetz

**Comment by mtarld** [comment]
Hi @art-cg !

Thanks for working hard on this bundle, I'm really grateful for it!

I began to work on that bundle because of a project I was working on about a year ago.
Actually, I do not work anymore on a project that involves API Platform in a microservice context, therefore except for dealing with technical debt, I cannot find use cases that the bundle is not covering.
That's why I don't have any roadmap yet and that's why I have to rely on contributors that have real needs like you (thanks again!)

Long story short, the roadmap is actually driven by users/contributors like you. Feel free to express your needs :slightly_smiling_face: 

But about the near future, I think we may:
-  Handle `PATCH` (#53)
- Log requests (#47)
And tag `0.4.0`

Then, we may:
- Be able to support Symfony 6.x
- Be able to support API Platform 2.7/3.0
- Deal with some technical debt
And tag `0.4.x`

**Comment by art-cg** [comment]
We have to thank you for this bundle.
Manageing microservices is not so easy as it sounds ;)

Well i was hopeing that you still habe this bundle in action. I can understand your motivation.

My personal roadmap is to finish the 2 PRs i started.
If i find the time i will try to solve Symfony 6 blocker in Api-Platform as linked in #61
After reaching a stable state, i want to move on to php8.
[Stats from packagist](https://packagist.org/packages/mtarld/api-platform-ms-bundle/php-stats#all) show that there is only php 7.4 and php 8 are useing it.
Maybe a 1.0 and remove the marked deprecated functions (as noted in the comments).
So the burden of maintaining is not to high.

Ideas for improving

- Auto-Discover the supported formats from a apip-service
- Define/Manage services globally (Microservice-Pool). We use a Bundle that is integrated in every service with shared objects.
- Additional ExistsConstraint with just the value saved. The IRI is saved in the Annotation (But the IRI concept is of course better)
- Improve the docs

I am not sure if it is the wort the effort and these are just some thoughts i have in mind.

**Comment by mtarld** [comment]
Sounds good!

About the PHP8 stuff, I cannot decide at the moment.
On one hand, I prefer using PHP8 and encourage other developers to do the same.
On the other hand, we are so sticking with API Platform that I like to have the exact same dependency. In that way, if someone can use API Platform, he can use that bundle also. 
AFAIK, it'll be the case for the 3.x versions of API Platform and 6.x of Symfony. Therefore, as soon as we drop the support for older versions, we'll be able to let the old stuff go :crossed_fingers: 
But we also can get ahead of Symfony and API Platform for a little time.
I think we must take a step back before tagging the 1.0 version to know what we wanna support.

About your other ideas, they are great and we need to go a little deeper on these. I think that you can split them into issues that we can schedule, WDYT?

**Comment by art-cg** [comment]
This sounds reasonable and good. I am fine with this.
I will make issues for enhancements as far i know more about the needs of our company. There are many moving parts and decision to make.
